src/content/en/updates/2021/02/faster-devtools-startup.md

```diff
@@ -0,0 +1,123 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Reduce DevTools performance overhead of message dispatch in the front-end.
+
+{# wf_updated_on: 2021-02-04 #}
+{# wf_published_on: 2021-02-04 #}
+{# wf_tags: devtools-blog #}
+{# wf_featured_image: /web/updates/images/2021/02/faster/devtools-blog.png #}
+{# wf_featured_snippet: Reduce DevTools performance overhead of message dispatch in the front-end. #}
+{# wf_blink_components: N/A #}
+
+# Improving DevTools startup time {: .page-title }
+
+{% include "web/_shared/contributors/sadym.html" %}
+
+<<../../_shared/devtools-research.md>>
+
+## DevTools startup now is ~13% faster 🎉 (from 11.2s down to 10s).
+TL;DR; The result is achieved by removing a redundant serialization.
+
+## Overview
+While DevTools is starting up, it needs to make some calls to the [V8 JavaScript engine V8](https://v8.dev/). 
+
+![DevTools starting up process](/web/updates/images/2021/02/faster/faster-1.svg)
+
+The mechanism Chromium uses to send DevTools commands to V8 (and for IPC in general) is called [`mojo`](https://chromium.googlesource.com/chromium/src/+/master/mojo/README.md). My teammates [Benedikt Meurer](https://twitter.com/bmeurer) and [Sigurd Schneider](/web/resources/contributors#sigurd-schneider) discovered an inefficiency while working on another task, and came up with an idea to improve the process by removing two redundant steps in how these messages are sent and received.
+
+Let us dive into how the `mojo` mechanism works!
+
+## The `mojo` mechanisms
+
+![The mojo mechanisms](/web/updates/images/2021/02/faster/faster-2.svg)
+
+There is a mojo command `EvaluateScript` which runs the JS command. It serializes the whole JS command including the `arguments` into a string of JavaScript source code that can be `eval()`’. As you might imagine, these strings can become quite long and expensive. After the command is received by V8, these strings of JavaScript code are deserialized before running. This process of serializing and deserializing for every single message creates significant overhead.
+
+Benedikt Meurer realised that serialisation and deserialisation of the `arguments` is quite expensive, and that the whole  **"Serialize JS command to JS string"** and **"Deserialize JS string"** steps are redundant and can be skipped. 
+
+Technical details: [`RenderFrameHostImpl::ExecuteJavaScript](https://source.chromium.org/chromium/chromium/src/+/master:content/browser/renderer_host/render_frame_host_impl.cc;drc=df872ce8fcce25af51aa6b0f9fe8b1135b687524;l=1677)
+
+## How we improved
+
+![Improved mechanisms](/web/updates/images/2021/02/faster/faster-3.svg)
+
+We introduced another mojo API method which allows us to pass the object name, the method to be called, and the list of arguments directly, instead of having to create the string of JavaScript source code. This allows us to skip serialization, deserialization, and the need to parse the JavaScript code.
```

Comment:
```suggestion
We introduced another mojo API method which allows us to pass the object name, the method to be called, and the list of arguments directly, instead of having to create the string of JavaScript source code. This allows us to skip serialization & deserialization, and removes the need to parse the JavaScript code.
```